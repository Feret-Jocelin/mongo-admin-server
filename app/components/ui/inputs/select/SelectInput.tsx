import styles from './select.module.scss';

import SvgIcon from '@components/ui/icon/SvgIcon';

interface SelectInputProps {
  label: string;
  value: string;
  // eslint-disable-next-line no-unused-vars
  onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
  options: string[];
  error?: string;
}

const SelectInput = ({ label, value, onChange, options, error }: SelectInputProps) => {
  return (
    <div className={styles.container}>
      <label>{label}</label>
      <select
        className={`${styles.select} ${error ? styles.selectError : ''}`}
        value={value}
        onChange={onChange}
      >
        {options.map((option) => (
          <option key={option} value={option}>{option}</option>
        ))}
      </select>
      <div className={styles.iconContainer}>
        <SvgIcon icon_name='caretUpDown' />
      </div>

      {error && <p className={styles.error}>{error}</p>}
    </div>
  );
}

export default SelectInput;