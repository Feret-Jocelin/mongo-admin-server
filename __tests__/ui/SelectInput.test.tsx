/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import SelectInput from '@/app/[locale]/components/ui/inputs/select/SelectInput';

const mock = {
  label: 'form Select',
  value: 'form Select value',
};
const optionsLanguages=  [
  { value: 'fr', label: 'FranÃ§ais', },
  { value: 'en', label: 'Anglais', },
  { value: 'es', label: 'Espagnol', },
];

describe('SelectInput', () => {
  it('renders', () => {
    const handleChange = jest.fn();
    const { container } = render(
      <SelectInput 
        label={mock.label}
        value={mock.value}
        options={optionsLanguages}
        onChange={handleChange} 
      />
    );
    expect(container).toMatchSnapshot();
  })
})