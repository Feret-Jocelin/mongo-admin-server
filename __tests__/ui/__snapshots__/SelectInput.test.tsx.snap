// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput renders 1`] = `
<div>
  <div
    class="container"
  >
    <label>
      form Select
    </label>
    <select
      class="select "
    >
      <option
        disabled=""
        value=""
      >
        Choisir
      </option>
      <option
        value="fr"
      >
        Fran√ßais
      </option>
      <option
        value="en"
      >
        Anglais
      </option>
      <option
        value="es"
      >
        Espagnol
      </option>
    </select>
    <div
      class="iconContainer"
    >
      <div
        class="undefined "
      />
    </div>
  </div>
</div>
`;
